app/controllers/admin/badge_achievements_controller.rb

```diff
@@ -0,0 +1,44 @@
+module Admin
+  class BadgeAchievementsController < Admin::ApplicationController
+    layout "admin"
+
+    def index
+      @badge_achievements = BadgeAchievement.all.page(params[:page]).order(created_at: :desc).per(10)
+    end
+
+    def destroy
+      @badge_achievement = BadgeAchievement.find(params[:id])
+
+      if @badge_achievement.destroy
+        flash[:success] = "Badge has been deleted!"
+      else
+        flash[:danger] = @badge_achievement.errors_as_sentence
+      end
+      redirect_to admin_badge_achievements_path
+    end
+
+    def award
+      @badge = Badge.all
```

Comment:
We don't have as many badges currently but we want this all to be very scalabe so we should try and avoid using `.all` whenever possible and use SQL paging